### ISSUE TITLE
Add post-processing methods that can be used across plugins

### ISSUE THREAD

**Issue by yaseminbridges** [monarch-initiative/pheval#50]
Methods to create the current pheval tsv output after post-processing can be implemented here to be used across various different plugins (Exomiser/Phen2Gene) to reduce repeated code 

- [x] #51
- [x] ~~Implement methods for adding ENSEMBL identifier to results if not already provided in tool-specific output~~
- [x] #53
- [x] #55

**Comment by matentzn** [comment]
I think I understand, but can you add a bit more detail to the ticket, like postprocessing of what, which postprocessing steps and which files are being processed? For context! :) Thanks!

**Comment by yaseminbridges** [comment]
No problem! :)

The idea is that the bare minimum data, which is the gene/variant data and score, from the tool-specific outputs are defined into dataclasses. As this is the current minimal data required to produce the PhEval tsv output these dataclasses should be defined in this repo to be used across different plugins.

The output of different tools are in various different outputs, with one example that Exomiser outputs json and Phen2Gene provides tsv so the methods in which I am iterating over the tool-specific outputs to define the variables to the dataclass will be different and contained to the specific plugin repo.

Addition of the ENSEMBL identifier to the results is a common method that should be used across plugins and can be put here - the analysis methods currently work by first attempting to match the ENSEMBL identifier and then the gene symbol to try and minimise the issues with changing gene data. 

The ranking or re-ranking of the results based on the score of interest should be the same across all tools - with equal scores being assigned an equal rank. This step should be put in the main repo as I found I repeated this in both the Exomiser and Phen2Gene plugins.

Final steps for producing the final data frame for tsv output can also be put in this repo as it was repeated. 

As a side note, the post-processing for gene and variant results are completely separated - which involves the production of separate PhEval gene and variant tsv outputs for Exomiser. The implementation of one without the other should not create any issues, i.e. Phen2Gene only performs gene prioritisation and so these results are post-processed to the PhEval gene tsv output with no issue

The steps for this are to:

- [ ] Define dataclasses for the bare minimum data required in the tool-specific output for variant and gene prioritisation results
- [ ] Implement methods for adding ENSEMBL identifier to results if not already provided in tool-specific output
- [ ] Implement methods for assigning ranks to result output
- [ ] Implement methods for producing the final PhEval gene/variant tsv output
